Library/Homebrew/os/linux/elf.rb

```diff
@@ -68,14 +68,52 @@ def binary_executable?
     elf_type == :executable
   end
 
+  def rpath
+    return @rpath if defined? @rpath
+
+    @rpath = if HOMEBREW_PATCHELF_RB
+      patcher = patchelf_patcher
+      begin
+        begin
```

Comment:
Can this use a single `begin` somehow?

Reply:
It can be we'll have to use inline rescue which violates rubocop rules.

Reply:
why does it need to rescue and then rescue the same exception again? It seems avoidable

Reply:
the outer "MissingTagException" is for `.rpath` call. 
Having a single rescue won't catch the second exception thrown.
i.e, the code in first rescue block can throw an exception the outer rescue handles that case.

Reply:
My eyes also went buggy reviewing this code. I suggest factoring out `rpath_patchelf` and `rpath_patchelf_rb`, and then also factoring out `runpath_patchelf_rb` (which should return `nil` when `DT_RUNPATH` is not found in the ELF) and call `runpath_patchelf_rb` from `rpath_patchelf_rb`.